caf-audit/src/test/java/com/hpe/caf/auditing/elastic/ElasticAuditIT.java

```diff
@@ -0,0 +1,337 @@
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_LONG_PARAM_FIELD,CUSTOM_DOC_LONG_PARAM_FIELD, docLongParamValue);
+            float docFloatParamValue = rand.nextFloat();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_FLOAT_PARAM_FIELD,CUSTOM_DOC_FLOAT_PARAM_FIELD, docFloatParamValue);
+            double docDoubleParamValue = rand.nextDouble();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_DOUBLE_PARAM_FIELD,CUSTOM_DOC_DOUBLE_PARAM_FIELD, docDoubleParamValue);
+            boolean docBooleanParamValue = rand.nextBoolean();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_BOOLEAN_PARAM_FIELD,CUSTOM_DOC_BOOLEAN_PARAM_FIELD, docBooleanParamValue);
+            Date docDateParamValue = new Date();
+            auditEventBuilder.addEventParameter(CUSTOM_DOC_DATE_PARAM_FIELD,CUSTOM_DOC_DATE_PARAM_FIELD, docDateParamValue);
+
+            //  Send audit event message to Elasticsearch.
+            auditEventBuilder.send();
+            Thread.sleep(1000);
+
+            //  Search Elasticsearch for the newly indexed document (by UserId).
+            try ( TransportClient transportClient = ElasticAuditTransportClientFactory.getTransportClient(esHostNames, ES_PORT, ES_CLUSTERNAME)) {
+                SearchHit[] hits = searchDocument(transportClient, ES_INDEX, ES_TYPE, USER_ID_FIELD, USER_ID);
+
+                //  Expecting a single hit.
+                Assert.assertTrue(hits.length == 1);
+
+                //  Verify fixed field data results.
+                verifyFixedFieldResult(hits, APP_ID_FIELD, APPLICATION_ID);
+                verifyFixedFieldResult(hits, EVENT_CATEGORY_ID_FIELD, EVENT_CATEGORY_ID);
+                verifyFixedFieldResult(hits, EVENT_TYPE_ID_FIELD, EVENT_TYPE_ID);
+                verifyFixedFieldResult(hits, USER_ID_FIELD, USER_ID);
+                verifyFixedFieldResult(hits, CORRELATION_ID_FIELD, CORRELATION_ID);
+
+                //  Verify fixed field data results.
+                verifyCustomFieldResult(hits, CUSTOM_DOC_STRING_PARAM_FIELD, docStringParamValue, "string");
+                verifyCustomFieldResult(hits, CUSTOM_DOC_INT_PARAM_FIELD, docIntParamValue, "int");
+                verifyCustomFieldResult(hits, CUSTOM_DOC_SHORT_PARAM_FIELD, docShortParamValue, "short");
+                verifyCustomFieldResult(hits, CUSTOM_DOC_LONG_PARAM_FIELD, docLongParamValue, "long");
+                verifyCustomFieldResult(hits, CUSTOM_DOC_FLOAT_PARAM_FIELD, docFloatParamValue, "float");
+                verifyCustomFieldResult(hits, CUSTOM_DOC_DOUBLE_PARAM_FIELD, docDoubleParamValue, "double");
+                verifyCustomFieldResult(hits, CUSTOM_DOC_BOOLEAN_PARAM_FIELD, docBooleanParamValue, "boolean");
+                verifyCustomFieldResult(hits, CUSTOM_DOC_DATE_PARAM_FIELD, docDateParamValue, "date");
+
+                //  Delete test document after verification is complete.
+                deleteDocument(transportClient, ES_INDEX, USER_ID_FIELD, USER_ID);
```

Comment:
In the same way that index can take time, delete might also.

This could cause a later test to find a document from an earlier one. So try adding unqiue data to the event as suggested above.